File: src/dotnet/HelloMutation.Domain/Entities/Author.cs
using System;

namespace HelloMutation.Domain.Entities
{
    public record Author
    {
        public Author(string firstName, string lastName) => (FirstName, LastName) = (firstName, lastName);

        private string _firstName;
        public string FirstName
        {
            get => _firstName;
            init => _firstName = !string.IsNullOrWhiteSpace(value) ? value : throw new ArgumentNullException(nameof(FirstName));
        }

        private string _lastName;
        public string LastName
        {
            get => _lastName;
            init => _lastName = !string.IsNullOrWhiteSpace(value) ? value : throw new ArgumentNullException(nameof(LastName));
        }
    }
}

File: src/javascript/src/domain/entities/author.ts
import { ArgumentNullError } from '../errors/argument_null_error'

export class Author {
    
    public firstName!: string
    public lastName!: string

    constructor(init: Partial<Author>) {
        if (!init.firstName) throw new ArgumentNullError('firstName')
        if (!init.lastName) throw new ArgumentNullError('lastName')
        Object.assign(this, init)
    }
}